Code/GraphMol/Depictor/Templates.cpp

```diff
@@ -0,0 +1,90 @@
+//
+//  Copyright (C) 2023 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "Templates.h"
+
+namespace RDDepict {
+bool CoordinateTemplates::isValidTemplate(RDKit::ROMol& mol, const std::string& smiles) {
+    // template must have 2D coordinates
+    if (mol.getNumConformers() == 0) {
+        BOOST_LOG(rdWarningLog) << "WARNING: Template is missing coordinates: "
+                                << smiles << "\n";
+        return false;
+    }
+    if (mol.getConformer().is3D()) {
+        BOOST_LOG(rdWarningLog) << "WARNING: Template has 3D coordinates, 2D "
+                                << "coordinates required: " << smiles << "\n";
+        return false;
+    }
+
+    // Make sure this template is a single ring system (spiro'd ring systems are OK).
+    // We can check this by ensuring that every atom is in a ring and that there
+    // is only one connected component in the molecular graph.
+    if (RDKit::MolOps::getMolFrags(mol).size() != 1) {
+        BOOST_LOG(rdWarningLog) << "WARNING: Template consists of multiple fragments, "
+                                << "single fragment required: " << smiles << "\n";
+        return false;
+    }
+
+    // Use symmetrizeSSSR since that is what is used in coordinate generation
+    bool includeDativeBonds = true;
+    RDKit::MolOps::symmetrizeSSSR(mol, includeDativeBonds);
+    auto ri = mol.getRingInfo();
+    for (unsigned int i = 0; i < mol.getNumAtoms(); ++i) {
+        if (!ri->numAtomRings(i)) {
+            BOOST_LOG(rdWarningLog) << "WARNING: Template is not a ring system: "
+                                    << smiles << "\n";
+            return false;
+        }
+    }
+    return true;
+}
+
+
+bool CoordinateTemplates::setRingSystemTemplates(const std::string& template_dir) {
+    std::ifstream cxsmiles(template_dir);
```

Comment:
template_dir seems a little misleading?

Reply:
good point, I'll rename this to `templatePath`